src/core/mempool.rs

```diff
@@ -1832,19 +1844,59 @@ impl MemPoolDB {
             num_written += nw;
 
             if query.tx_buf_ptr >= query.tx_buf.len() {
+                if query.corked {
+                    // we're done
+                    test_debug!(
+                        "Finished streaming txs; last page was {:?}",
+                        &query.last_randomized_txid
+                    );
+                    break;
+                }
+
+                if query.num_txs >= query.max_txs {
+                    // no more space in this stream
+                    test_debug!(
+                        "No more space in this query after {:?}",
+                        &query.last_randomized_txid
+                    );
+
+                    // send the next page ID
+                    query.tx_buf_ptr = 0;
+                    query.tx_buf.clear();
+                    query.corked = true;
+
+                    test_debug!(
+                        "Cork tx stream with next page {}",
+                        &query.last_randomized_txid
+                    );
```

Comment:
Can this `test_debug!` be a `debug!` log, and combined with the `test_debug!` log line at 1858 -- this could be useful in debugging a running node's behavior, but it seems like they can be one log line rather than two.

Reply:
Yup, fixed